recipe/build_folly.sh

```diff
@@ -50,12 +52,27 @@ if [[ "${target_platform}" == "linux-ppc64le" ]]; then
     CMAKE_ARGS="${CMAKE_ARGS} -DHAVE_VSNPRINTF_ERRORS_EXITCODE=0 -DHAVE_VSNPRINTF_ERRORS_EXITCODE__TRYRUN_OUTPUT=''"
 fi
 
-cmake ${CMAKE_ARGS} -Wno-dev -GNinja -DBUILD_SHARED_LIBS=ON ..
+# Build a shared library for the "folly" package or
+# build a static library for the "folly-static" package.
+if [[ ! -z "${PKG_NAME+x}" && "${PKG_NAME}" == "folly" ]]
+then
+    CMAKE_ARGS="${CMAKE_ARGS} -DBUILD_SHARED_LIBS=ON"
+else
+    CMAKE_ARGS="${CMAKE_ARGS} -DBUILD_SHARED_LIBS=OFF"
+fi
+
+echo "a) THE PREFIX" $PREFIX 
+echo "a) THE BUILD PREFIX" $BUILD_PREFIX
+
+$BUILD_PREFIX/bin/cmake ${CMAKE_ARGS} -Wno-dev -GNinja ..
+
+echo "b) THE PREFIX" $PREFIX 
+echo "b) THE BUILD PREFIX" $BUILD_PREFIX
 
 cat CMakeCache.txt
 
-cmake --build . --parallel
+$BUILD_PREFIX/bin/cmake --build . --parallel 
```

Comment:
This should not be necessary. If a different CMake (not the one in $BUILD_PREFIX) is picked up, this is an indication that something else in your environment is broken.
```suggestion
cmake --build . --parallel 
```